official/transformer/v2/transformer_main.py

```diff
@@ -124,31 +123,49 @@ def train(self):
     valid_ds = valid_ds.map(
         map_data_fn, num_parallel_calls=params["num_parallel_calls"])
 
-    init_epoch = flags_obj.init_epoch or 0
-    init_steps = init_epoch * flags_obj.steps_per_epoch
-    callbacks = self._create_callbacks(cur_log_dir, init_steps, params)
-
-    history = model.fit(
-        train_ds,
-        initial_epoch=init_epoch,
-        epochs=flags_obj.train_epochs,
-        steps_per_epoch=flags_obj.steps_per_epoch,
-        validation_data=valid_ds,
-        validation_steps=flags_obj.validation_steps,
-        callbacks=callbacks)
-    tf.compat.v1.logging.info("\nTrain history: {}".format(history.history))
-
-    save_weight_path = os.path.join(cur_log_dir, "saves-model-weights.hdf5")
-    save_model_path = os.path.join(cur_log_dir, "saves-model.hdf5")
-    model.save_weights(save_weight_path)
-    model.save(save_model_path)
+    callbacks = self._create_callbacks(flags_obj.model_dir, 0, params)
+
+    epochs = 0
+    predict_model = None
+    for i in xrange(1, flags_obj.train_epochs + 1):
+      print("Train Epoch:{}/{}".format(i, flags_obj.train_epochs))
+      history = model.fit(
+          train_ds,
+          initial_epoch=epochs,
+          epochs=i,
+          steps_per_epoch=flags_obj.train_steps,
+          callbacks=callbacks,
+          verbose=2)
+      epochs += 1
```

Comment:
this can be just `i`?

Reply:
Sorry about that, much more streamlined now.